src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
-export interface Figure {
 
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Colors {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape = Shapes.Triangle;
 
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
```

Comment:
You can check it by getting largest side with Math.max(a, b, c) and calculating only once:
(a + b + c - largest) <= largest. This is shorted your code :)